gordon/interfaces.py

```diff
@@ -73,52 +88,26 @@ def __init__(config, success_channel, error_channel, metrics=None):
             metrics (obj): Optional obj used to emit metrics.
         """
 
-    async def shutdown():
-        """Gracefully shutdown plugin."""
-
-
-class IEventConsumerClient(IGenericPlugin):
-    """Client for ingesting push/pull events for Gordon to process.
-
-    The client also receives both successful and failed
-    :py:class:`gordon.interfaces.IEventMessage` objects from Gordon in order
-    to perform cleanup if needed.
-    """
-
     async def run():
         """Begin consuming messages using the provided event loop."""
 
-    async def cleanup(event_msg):
-        """Perform cleanup tasks related to a message.
-
-        Args:
-            event_msg (IEventMessage): Message at the end of its life cycle.
-        """
 
+class IMessageHandler(IGenericPlugin):
+    """Plugin which performs some operation on an event message."""
 
-class IEnricherClient(IGenericPlugin):
-    """Client for processing (enriching) or filtering events received by Gordon.
-
-    Note that if no extra processing is required, the implementer can
-    immediately push the received event into the success_channel.
-    """
-
-    async def process(event_msg):
-        """Process message.
+    def __init__(config, metrics, **kwargs):
```

Comment:
The "event consumer"-type message handler is a special case, which should get the success channel, since we're expecting something to consume from an external queue to produce messages for the internal queue.

One way is to have a special interface that inherits from both this and runnable. I'm not necessarily married to that. But perhaps _for now_ it'd be enough to just make it clear in prose docs that we expect an implementation that implements both runnable and message handler in order to produce messages for the router to then route.

Reply:
I thought about this, too, especially since I didn't like the conflicting init methods, but wasn't sure since we're not verifying yet. I'm fine with adding it, though.